#!/usr/bin/env cwl-runner
cwlVersion: v1.2
class: Workflow
label: "Subworkflow for Annotation"
doc: |
  "Subworkflow for Metagenome Annotation
  This subworkflow is for annotation of predicted protein coding sequences.
  "

requirements:
  # https://www.commonwl.org/user_guide/topics/workflows.html#scattering-steps
  ScatterFeatureRequirement: {}


inputs:
############## Common Parameters in this subworkflow ##############
  - id: SW_THREADS
    type: int
    label: "threads"
    doc: "number of threads to use in this subworkflow"
    default: 16

  - id: SW_EVALUE
    type: float
    label: "evalue"
    doc: "E-value threshold of BLASTP (diamond) and BLASTN alignment"
    default: 0.1


############## Parameters for creating diamond index (/Tools/09_diamond_index.cwl) ##############
  - id: SW_UNIPROT_PROTEIN_FASTA_FILE
    type: File
    label: "uniprot protein fasta file"
    doc: "uniprot protein fasta file"
    default:
      class: File
      location: ../Data/uniprot_sprot.fasta
  

############## Parameters for creating rRNA index and BLASTn alignment (/Worlkflow/blastn_rRNA_scatter.cwl) ##############
  - id: SW_BLASTN_rRNA_FASTA_FILE1
    type: File
    label: "Si"
    doc: "SILVA_138.1_LSUParc_tax_silva"
    default:
      class: File
      location: ../Data/SILVA_138.1_LSUParc_tax_silva.fasta.gz
  
  - id: SW_BLASTN_rRNA_FASTA_FILE2
    type: File
    label: "SILVA_138.1_SSUParc_tax_silva"
    doc: "SILVA_138.1_SSUParc_tax_silva"
    default:
      class: File
      location: ../Data/SILVA_138.1_SSUParc_tax_silva.fasta.gz

  - id: SW_INPUT_FASTA_FILE
    type: File
    label: "input fasta file (nucleotide sequence generated by prodigal process)"
    doc: "predicted protein coding sequences produced by Prodigal process"
    default:
      class: File
      location: ../out/all_contigs_SRR27548858_dna.fasta
  
############## Parameters for diamond alignment (/Tools/11_diamond_uniprot_alignment.cwl) ##############
  - id: 



steps:
  - id: mapping 